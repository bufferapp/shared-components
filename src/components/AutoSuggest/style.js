/* eslint-disable no-nested-ternary */
import styled from 'styled-components';
import {
  grayShadow,
  grayLighter,
  white,
  gray,
  grayLight,
  grayDarker,
  redLight,
  red,
} from '../style/colors';
import {
  fontFamily,
  fontSize,
  fontWeight,
  lineHeight
} from '../style/fonts';

export const Wrapper = styled.div`
  outline: 0;
  :focus {
    outline: 0;
  }
  width: ${props => (props.isSplit || props.fullWidth ? '100%' : 'auto')};
  height: ${props => (props.isSplit ? '100%' : 'auto')};
  max-height: ${props => (props.isSplit ? '100%' : '')};
  position: ${props => (props.isSplit ? 'initial' : 'relative')};
  display: inline-block;
`;

export const SelectStyled = styled.div`
  right: ${props => (props.hasIconOnly ? '6px' : '0')};
  position: absolute;
  z-index: 1000;
  border: 1px solid ${gray};
  box-sizing: border-box;
  box-shadow: ${grayShadow};
  border-radius: 4px;
  display: ${props => (props.isOpen ? 'initial' : 'none')};
  min-width: 200px;
  max-width: ${props => (props.fullWidth ? '' : '200px')};
  width: ${props => (props.fullWidth ? '100%' : '200px')};
  background-color: #ffffff;
  bottom: ${props => (props.yPosition === 'top' ? '100%' : 'initial')};
  top: ${props => (props.yPosition === 'bottom' ? '100%' : 'initial')};
  margin-bottom: ${props => (props.yPosition === 'top' ? '8px' : '0')};
  margin-top: ${props =>
    props.marginTop
      ? props.marginTop
      : props.yPosition === 'bottom' || props.xPosition === 'right'
      ? '8px'
      : '0'};
  transform: translateX(
    ${props => (props.horizontalOffset ? props.horizontalOffset : '0')}
  );
  left: ${props => (props.xPosition === 'left' ? 0 : '')};
`;

export const SelectItems = styled.ul`
  max-height: 195px;
  overflow-y: auto;
  padding-bottom: 16px;
  padding-top: 16px;
  position: relative;
  z-index: -1;
  background: ${white};
  margin-block-start: 0px;
  margin-block-end: 0px;
  padding-inline-start: 0px;
  border-radius: 4px;
  list-style: none;
`;

export const SelectItemDivider = styled.li`
  background-color: ${grayLighter};
  height: 1px;
  margin-bottom: 8px;
  margin-top: 8px;
  padding: 0;
  pointer-events: none;
  width: 100%;
`;

export const SelectItemDividerTitle = styled.span`
  position: relative;
  top: -11px;
  margin-left: 8px;
  background: white;
  padding: 0 8px;
  font-size: 12px;
  font-family: ${fontFamily};
  color: ${gray};
`;

export const SearchBarWrapper = styled.div`
  display: flex;
  box-sizing: border-box;
  margin: 8px 0px;
  background: ${({ hasError }) => (hasError ? redLight : white)};
  border-radius: 4px;
  border: 1px solid ${({ hasError }) => (hasError ? red : grayLight)};
  box-shadow: 2px 2px 0 2px transparent;
  transition-property: border-width, border-color, box-shadow;
  transition-duration: 0.1s;
  transition-timing-function: ease-in;
  color: ${({ hasError }) => (hasError ? red : grayDarker)};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${fontWeight};
  line-height: ${lineHeight};
  padding: ${({ size }) => {
  switch (size) {
    case 'small':
      return '5px 8px 4px 8px';
    case 'tall':
      return '13px 16px 12px 16px';
    default:
      return '9px 8px 8px 8px';
  }
}};
  width: 100%;
  color: ${gray};
`;

export const NoMatchesContainer = styled.p`
  width: 116px;
  height: 17px;
  margin: 0 auto;
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  line-height: normal;
  font-size: 14px;
  letter-spacing: -0.233333px;
  color: ${gray};
`;

export const CustomItemContainer = styled.li`
  min-height: 32px;
  align-items: center;
  color: ${grayDarker};
  cursor: pointer;
  display: flex;
  font-size: 14px;
  font-weight: 500;
  overflow: hidden;
  text-overflow: ellipsis;
  user-select: none;
  white-space: nowrap;
  margin-left: 8px;
  margin-right: 8px;
  border-radius: 4px;
  padding-left: 8px;
  padding-right: 8px;
  &:hover {
    background-color: ${grayLight};
  }
  background-color: ${props =>
    props.isCustomItemFocused ? `${grayLight}` : ''};
`;
