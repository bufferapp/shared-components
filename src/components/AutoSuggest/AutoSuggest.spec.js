import snap from 'jest-auto-snapshots';
import AutoSuggest from './AutoSuggest';

snap(AutoSuggest, './AutoSuggest.jsx');
