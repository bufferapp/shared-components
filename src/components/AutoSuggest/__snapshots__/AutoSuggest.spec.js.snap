// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when array prop "hotKeys" has one item: "[object Object]" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when array prop "hotKeys" is an empty array: "[]" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when array prop "items" has one item: "[object Object]" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "disabled" is set to: "false" 1`] = `
<styled.div
  data-tip="jest-auto-snapshots String Fixture"
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "fullWidth" is set to: "false" 1`] = `
<styled.div
  data-tip=""
  fullWidth={false}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={false}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "hasCustomAction" is set to: "false" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "hasIconOnly" is set to: "false" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={false}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "hideSearch" is set to: "false" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={false}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "isOpen" is set to: "false" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={false}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "isSplit" is set to: "false" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={false}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "multiSelect" is set to: "false" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={false}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "selectPopupVisible" is set to: "false" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={false}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when boolean prop "shortcutsEnabled" is set to: "false" 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when passed all props 1`] = `
<styled.div
  data-tip=""
  fullWidth={true}
  innerRef={[Function]}
  isSplit={true}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="jest-auto-snapshots String Fixture"
    />
  </styled.div>
  <styled.div
    fullWidth={true}
    hasIconOnly={true}
    horizontalOffset="jest-auto-snapshots String Fixture"
    isOpen={true}
    marginTop="jest-auto-snapshots String Fixture"
    xPosition="right"
    yPosition="top"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={true}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        keyMap={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        multiSelect={true}
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > AutoSuggest Matches snapshot when passed only required props 1`] = `
<styled.div
  innerRef={[Function]}
  isSplit={false}
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <styled.div
    id="searchInput"
    innerRef={[Function]}
  >
    <SearchIcon />
    <Search
      height="small"
      innerRef={[Function]}
      isOpen={true}
      onChange={[Function]}
      onClick={[Function]}
      placeholder="Search"
    />
  </styled.div>
  <styled.div
    hasIconOnly={false}
    isOpen={false}
    xPosition="left"
    yPosition="bottom"
  >
    <styled.ul
      innerRef={[Function]}
    >
      <SelectItem
        getItemId={[Function]}
        hasSelectedItems={false}
        hideSearch={false}
        hovered={false}
        item={
          Object {
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          }
        }
        onClick={[Function]}
      />
    </styled.ul>
  </styled.div>
</styled.div>
`;
